lib/cli/index.spec.js

```diff
@@ -1,137 +1,58 @@
-    jest.spyOn(JSON, 'parse');
+    console.warn.mockImplementation(v => v);
   });
 
   beforeEach(() => {
-    argv = { files: ['config.json'] };
-
+    fs.readJson.mockReturnValue({ name: 'foo' });
     console.log.mockReset();
-
-    parse.mockReset();
-    parse.mockReturnValue(argv);
-
-    fs.read.mockClear();
-    fs.write.mockClear();
-
-    path.resolve.mockClear();
-    format.mockClear();
-
-    fs.read.mockReturnValue('{}');
-    format.mockReturnValue('next');
+    console.warn.mockReset();
   });
 
   afterAll(() => {
     console.log.mockRestore();
-    JSON.parse.mockRestore();
   });
 
   it('parses arguments', async () => {
     expect.assertions(1);
-    await cli('argv');
-    expect(parse).toHaveBeenCalledWith('argv');
-  });
 
-  it('resolves the path relative to process.cwd', async () => {
-    expect.assertions(1);
-    await cli('argv');
-    expect(path.resolve).toHaveBeenCalledWith('config.json');
-  });
-
-  it('loads the config', async () => {
-    expect.assertions(1);
-    argv.config = 'config.json';
-    await cli('argv');
-    expect(format).toHaveBeenCalledWith({}, config);
-  });
+    const configSpy = jest.spyOn(config, 'search');
+    await cli.execute('--verbose');
 
-  it('reads the contents of the previous package and calls format', async () => {
-    expect.assertions(3);
-
-    argv.config = 'config.json';
-
-    await cli('argv');
-
-    expect(fs.read).toHaveBeenCalledWith(argv.files[0], 'utf8');
-    expect(JSON.parse).toHaveBeenCalledWith('{}');
-    expect(format).toHaveBeenCalledWith({}, { configOptions: true });
+    expect(configSpy).toHaveBeenCalled();
   });
 
   it('writes the contents', async () => {
-    expect.assertions(2);
-
-    argv.write = true;
-
-    await cli('argv');
-
-    expect(fs.write).toHaveBeenCalledWith(argv.files[0], 'next', {
-      encoding: 'utf8',
-    });
-
-    expect(console.log).toHaveBeenCalledTimes(2);
+    expect.assertions(1);
+    await cli.execute('--write');
+    expect(fs.writeFile).toHaveBeenCalled();
   });
 
   it('prints the contents if verbose is set', async () => {
     expect.assertions(1);
-
-    argv.verbose = true;
-    argv.write = true;
-
-    await cli('argv');
-
-    expect(console.log).toHaveBeenCalledTimes(2);
-  });
-
-  it('prints the contents if write is not set', async () => {
-    expect.assertions(2);
-
-    argv.write = false;
-
-    await cli('argv');
-
+    await cli.execute('--verbose --write');
     expect(console.log).toHaveBeenCalledTimes(2);
-    expect(fs.write).not.toHaveBeenCalled();
   });
 
   it('catches errors', async () => {
-    expect.assertions(1);
-
-    const error = new Error('Foo');
-    fs.read.mockImplementation(() => {
-      throw error;
-    });
-
-    await cli('argv');
-
-    expect(logErrorAndExit).toHaveBeenCalledWith(error);
+    expect.assertions(2);
+    await cli.execute(null);
+    expect(logErrorAndExit).toHaveBeenCalled();
+    expect(console.warn).toHaveBeenCalledTimes(1);
   });
 });
```

Comment:
The addition of `cosmiconfig` simplified `index.spec.js` by delegating all config parsing tests to `config.spec.js`.